### DISCUSSION TITLE
Mesmerizer flashed from platformIO. No pixels lighting up.

### DISCUSSION THREAD

**Discussion by KeiranHines** [PlummersSoftwareLLC/NightDriverStrip#423] [Q&A]
Hi All, 

I think I have soft bricked my board and wanted to know how to recover it. The current state of the board is that I can Boot the board, it connects to my Wifi. I can connect to the frontend and the IR Remote triggers, but no pixels are lit up.
I was attempting to reflash from pio because the web installer was not working. After some messing around I was able to flash the board. 
After doing that the serial output listed my wifi, but then that was quickly replaced with an 'IoTWifi' SSID. (Logs below)
```
I) (PrintOutputHeader)(C1) Version 40: Wifi SSID: "<MY_SSID_IS_PRINTED_HERE>" - ESP32 Free Memory: 235060, PSRAM:4191883, PSRAM Free: 4187123
(I) (PrintOutputHeader)(C1) ESP32 Clock Freq : 240 MHz
(I) (setup)(C1) Startup!
(I) (setup)(C1) Starting DebugLoopTaskEntry
>> Launching Debug Thread.  Mem: 235060, LargestBlk: 110580, PSRAM Free: 4187123/4191883, >> Launching JSON Writer Thread.  Mem: 226492, LargestBlk: 110580, PSRAM Free: 4187087/4191867, (W) (DeviceConfig)(C1) DeviceConfig could not be loaded from JSON, using defaults
(W) (NotifyJSONWriterThread)(C1) >> Notifying JSON Writer Thread
(W) (setup)(C1) Starting ImprovSerial
(W) (ReadWiFiConfig)(C1) Retrieved SSID and Password from NVS: IoTWiFi, ********

Starting SmartMatrix Mallocs
Heap/32-bit Memory Available: 221340 bytes total, 110580 bytes largest free block
8-bit/DMA Memory Available  : 191044 bytes total, 110580 bytes largest free block
Total PSRAM used: 11328 bytes total, 4180459 PSRAM bytes free
SmartMatrix Layers Allocated from Heap:
Heap/32-bit Memory Available: 219864 bytes total, 110580 bytes largest free block
Starting SmartMatrix DMA Mallocs
sizeof framestruct: 00004000
DMA Memory Available before ptr1 alloc: 156768 bytes total, 110580 bytes largest free block
matrixUpdateFrames[0] pointer: 3FFD2B60
DMA Memory Available before ptr2 alloc: 156768 bytes total, 110580 bytes largest free block
matrixUpdateFrames[1] pointer: 3FFD6B70
Frame Structs Allocated from Heap:
Heap/32-bit Memory Available: 187064 bytes total, 110580 bytes largest free block
8-bit/DMA Memory Available  : 156768 bytes total, 110580 bytes largest free block
Total PSRAM used: 37312 bytes total, 4154347 PSRAM bytes free
Allocating refresh buffer:
lsbMsbTransitionBit of 0 requires 49152 RAM, 110580 available, leaving 61428 free:
Raised lsbMsbTransitionBit to 0/7 to fit in RAM
lsbMsbTransitionBit of 0 gives 76 Hz refresh, 180 requested:
lsbMsbTransitionBit of 1 gives 152 Hz refresh, 180 requested:
lsbMsbTransitionBit of 2 gives 300 Hz refresh, 180 requested:
Raised lsbMsbTransitionBit to 2/7 to meet minimum refresh rate
Descriptors for lsbMsbTransitionBit 2/7 with 16 rows require 12288 bytes of DMA RAM
SmartMatrix Mallocs Complete
Heap/32-bit Memory Available: 174744 bytes total, 110580 bytes largest free block
8-bit/DMA Memory Available  : 144448 bytes total, 110580 bytes largest free block
Total PSRAM used: 37312 bytes total, 4154347 PSRAM bytes free
Setting up parallel I2S bus at I2S1
Matrix Refresh Rate: 100
(I) (SetupBufferManagers)(C1) Could allocate 509 buffers but limiting it to 500
(I)
(W) (SetupBufferManagers)(C1) Reserving 500 LED buffers for a total of 3088000 bytes...
(I) (setup)(C1) Initializing splash effect manager...
(W) (InitSplashEffectManager)(C1) InitSplashEffectManager
(W) (InitEffectsManager)(C1) InitEffectsManager...
(I) (LoadJSONFile)(C1) Attempting to read JSON file /effects.cfg
(W) (LoadJSONFile)(C1) Out of memory reading JSON from file /effects.cfg - increasing buffer to 4931 bytes
(W) (LoadJSONFile)(C1) Out of memory reading JSON from file /effects.cfg - increasing buffer to 6979 bytes
(I) (InitEffectsManager)(C1) Creating EffectManager from JSON config
(I) (ReadCurrentEffectIndex)(C1) Attempting to read file /current.cfg
>> Launching Drawing Thread.  Mem: 173156, LargestBlk: 110580, PSRAM Free: 1010795/4174379, (W) (DrawLoopTaskEntry)(C1) >> DrawLoopTaskEntry
(W)
(W) (DrawLoopTaskEntry)(C1) Entering main draw loop!
>> Launching Audio Thread.  Mem: 168684, LargestBlk: 110580, PSRAM Free: 1010795/4174379, (I) (AudioSamplerTaskEntry)(C0) >>> Sampler Task Started
>> Launching Remote Thread.  Mem: 163988, LargestBlk: 110580, PSRAM Free: 1010627/4174363, (W) (begin)(C1) Remote Control Decoding Started
>> Launching Network Thread.  Mem: 157828, LargestBlk: 110580, PSRAM Free: 1010099/4174347, [  2127][E][ESPmDNS.cpp:65] begin(): Failed starting MDNS
Error starting mDNS
(W) (UpdateSubscribers)(C1) Skipping Subscriber update, waiting for WiFi...
>> Launching ColorData Thread.  Mem: 148972, LargestBlk: 110580, PSRAM Free: 1010291/4174347, >> Launching Socket Thread.  Mem: 146036, LargestBlk: 110580, PSRAM Free: 1010291/4174347, (W) (NotifyJSONWriterThread)(C1) >> Notifying JSON Writer Thread
(I) (ConnectToWiFi)(C1) Setting host name to NightDriverStrip...WL_NO_SHIELD
(W) (ConnectToWiFi)(C1) Pass 1 of 5: Connecting to Wifi SSID: "IoTWiFi" - ESP32 Free Memory: 108184, PSRAM:4174091, PSRAM Free: 997147
```

In an attempt to fix that I attempted to upload load a file system image and erase the flash. I assume that is where I went wrong. Now I just get the below output when monitoring. I can see it appears to be attempting to draw, but nothing is output of the LED matrix. 
```
--- Terminal on COM6 | 115200 8-N-1
--- Available filters and text transformations: colorize, debug, default, direct, esp32_exception_decoder, hexlify, log2file, nocontrol, printable, send_on_enter, time
--- More details at https://bit.ly/pio-monitor-filters
--- Quit: Ctrl+C | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H
(I) (ConnectToWiFi)(C1) Setting host name to NightDriverStrip...WL_NO_SHIELD
(W) (ConnectToWiFi)(C1) Pass 1 of 5: Connecting to Wifi SSID: "Shelbyville" - ESP32 Free Memory: 108124, PSRAM:4174091, PSRAM Free: 997675
(W) 
(W) (ColorDataTaskEntry)(C1) Started color data server!
(I) (SaveToJSONFile)(C0) Number of bytes written to JSON file /effects.cfg: 2883
(I) (loop)(C1) WiFi: WL_CONNECTED, IP: 10.1.5.1, Mem: 105360, LargestBlk: 73716, PSRAM Free: 989079/4174027, LED FPS: 25 Refresh: 60 Hz, Power: 1942 mW, Brite: 100%, Audio FPS: 43, MinVU: 1204.4, PeakVU: 1318.9, VURatio: 1.3 Buffer: 0/500, CPU: 093%, 068%, FreeDraw: 0.015
(W) (NotifyJSONWriterThread)(C1) >> Notifying JSON Writer Thread
(W) (ConnectToWiFi)(C1) Connected to AP with BSSID: 7C:26:64:20:20:DE
(W)
(W) (ConnectToWiFi)(C1) Received IP: 10.1.5.1
(I) (ConnectToWiFi)(C1) Starting/restarting Socket Server...
(W) (ProcessIncomingConnectionsLoop)(C1) Error accepting data!
(W) (SockettToWiFi)(C1) Socket server started.
sed.  Retrying...
(I)
I)
tToWiFi)(C1) Socket server started.
(I) (ConnectToWiFi)(C1) Publishing OTA...
(I) (ConnectToWiFi)(C1) Setting Clock...
[  7848][E][WiFiUdp.cpp:221] parsePacket(): could not receive data: 9
(W) (UpdateClockFromWeb)(C1) NTP clock: Raw values sec=2241654079, usec=521925
(I) (UpdateClockFromWeb)(C1) Old Time: 7.954573, New Time: 1694222180.521925, Delta: 1694222172.567352
(I) (UpdateClockFromWeb)(C1) Adjusting time by 1694222172.567352 to 1694222180.521925
(I) (UpdateClockFromWeb)(C1) NTP clock: response received, updated time to: 1694222180.521925, DELTA: 1694222172.567352
(I) 
(I) (ConnectToWiFi)(C1) Starting Web Server...
(I) (begin)(C1) Connecting Web Endpoints
(I) (begin)(C1) Embedded html file size: 1350
(I) (begin)(C1) Embedded jsx file size: 41450
(I) (begin)(C1) Embedded ico file size: 15406
(I) (begin)(C1) Embedded timezones file size: 17834
(I) (begin)(C1) HTTP server started
(I) (ConnectToWiFi)(C1) Web Server begin called!
(I) (WriteCurrentEffectIndexFile)(C0) Number of bytes written to file /current.cfg: 1
[ 11220][E][WiFiClient.cpp:517] flush(): fail on fd 54, errno: 11, "No more processes"
(I) (UpdateSubscribers)(C1) Got YouTube subscriber count for channel Daves Garage (GUID 9558daa1-eae8-482f-8066-17fa787bc0e4)
(I) (loop)(C1) WiFi: WL_CONNECTED, IP: 10.1.5.1, Mem: 75636, LargestBlk: 42996, PSRAM Free: 985135/4173915, LED FPS: 53 Refresh: 60 Hz, Power: 2017 mW, Brite: 100%, Audio FPS: 41, MinVU: 1035.4, PeakVU: 1062.3, VURatio: 0.1 Buffer: 0/500, CPU: 093%, 073%, FreeDraw: 0.004
(I) (loop)(C1) WiFi: WL_CONNECTED, IP: 10.1.5.1, Mem: 75636, LargestBlk: 42996, PSRAM Free: 985663/4173931, LED FPS: 56 Refresh: 60 Hz, Power: 1977 mW, Brite: 100%, Audio FPS: 45, MinVU:  639.8, PeakVU:  688.0, VURatio: 0.5 Buffer: 0/500, CPU: 075%, 065%, FreeDraw: 0.008
```

Any help would be greatly appreciated. 

Thanks, 
Keiran.

**Comment by robertlipe** [comment]
I'm away from my Mesmerizer for a few days. This would be a lovely chance
for one of our new members with a fresh Mesmerizer to help pick through the
logs of their devices and help compare notes.

On Fri, Sep 8, 2023 at 9:18 PM Keiran Hines ***@***.***>
wrote:

> I think I have soft bricked my board and wanted to know how to recover it.
> The
>
The good news is that WAY too much is working for your board to be
considered a brick. It's clearly running Mesmerizer code, connecting with
the host just fine and so on.

It's POSSIBLE that you may have blown/lost an LCD panel, a level shifter,
an I/O interface pin, or something else. But for now, let's think happy
thoughts and work through the logs.

> After doing that the serial output listed my wifi, but then that was
> quickly replaced with an 'IoTWifi' SSID. (Logs below)
>
That's odd. The word "IoTWiFi" doesn't really appear in the code anywhere
that I can find. Do you have any theories where that name came from?

(W) (ReadWiFiConfig)(C1) Retrieved SSID and Password from NVS: IoTWiFi,
********

So that name seems to be coming from your NonVolatile storage. Are you
really saying you never gave it that name?

> Starting SmartMatrix Mallocs
>
> Some of that may seem scary sounding, but from memory, it all seems about
right.

> (I) (LoadJSONFile)(C1) Attempting to read JSON file /effects.cfg
> (W) (LoadJSONFile)(C1) Out of memory reading JSON from file /effects.cfg - increasing buffer to 4931 bytes
> (W) (LoadJSONFile)(C1) Out of memory reading JSON from file /effects.cfg - increasing buffer to 6979 bytes
>
> This is actually normal.|

> >> Launching Network Thread.  Mem: 157828, LargestBlk: 110580, PSRAM Free: 1010099/4174347, [  2127][E][ESPmDNS.cpp:65] begin(): Failed starting MDNS
> Error starting mDNS
>
> Now this, I don't remember seeing on eery boot. Can anyone else comment on
whether they're seeing this one in a similar position?

> (W) (UpdateSubscribers)(C1) Skipping Subscriber update, waiting for WiFi...
> >> Launching ColorData Thread.  Mem: 148972, LargestBlk: 110580, PSRAM Free: 1010291/4174347, >> Launching Socket Thread.  Mem: 146036, LargestBlk: 110580, PSRAM Free: 1010291/4174347, (W) (NotifyJSONWriterThread)(C1) >> Notifying JSON Writer Thread
> (I) (ConnectToWiFi)(C1) Setting host name to NightDriverStrip...WL_NO_SHIELD
>
> Now THIS is concerning. This doesn't seem like an error, but is showing
that src/network.cpp::ConnectToWiFi() is having trouble talking to the
board. That would be super strange if the name actually came from DHCP,
which would mean the name came from thee etwork - that we now seemingly
can't find. It would be less strange if you had another theory of where
that name came from

This means that
        debugI("Setting host name to %s...%s",
cszHostname,WLtoString(WiFi.status()));

is being hit with a WiFi.status of WIFI_NO_SHIELD which sure looks like an
error condition and not not a happy state.

>
> (W) (ConnectToWiFi)(C1) Pass 1 of 5: Connecting to Wifi SSID: "IoTWiFi" - ESP32 Free Memory: 108184, PSRAM:4174091, PSRAM Free: 997147
>
> In an attempt to fix that I attempted to upload load a file system image
> and erase the flash. I assume that is where I went wrong. Now I just get
> the below output when monitoring. I can see it appears to be attempting to
> draw, but nothing is output of the LED matrix.
>
Describe how you erased the flash and how you recovered from that. Did you
upload both the app and user partitinos or just the app partition?

> (W) (ConnectToWiFi)(C1) Pass 1 of 5: Connecting to Wifi SSID: "Shelbyville" - ESP32 Free Memory: 108124, PSRAM:4174091, PSRAM Free: 997675
>
> I know it's not the point, but are you also in Middle TN? It would be cool
to have three of us....

(W) (ProcessIncomingConnectionsLoop)(C1) Error accepting data!
> (W) (SockettToWiFi)(C1) Socket server started.
> sed.  Retrying...
> (I)
> I)
> tToWiFi)(C1) Socket server started.
> (I) (ConnectToWiFi)(C1) Publishing OTA...
>
> This section is a bit funky. Here, it looks like you - or something else -
tried to start an over-the-air flash update while you were monitoring and
trying to load from COM6 (?) above.




> (I) (ConnectToWiFi)(C1) Setting Clock...
> [  7848][E][WiFiUdp.cpp:221] parsePacket(): could not receive data: 9
>
>
This is the WiFi over the air updater getting data that's corrupt.

It's my experience that mixing over the air and direct serial updates works
poorly.  Until you get your board back on track, I'd recommend using only
direct connection.

> (I) (UpdateClockFromWeb)(C1) NTP clock: response received, updated time to: 1694222180.521925, DELTA: 1694222172.567352
>
> That's interesting. You were getting OTA packetts just moment ago, but now
you're getting that first NTP response where the board yells out "Does
anybody really know what time it is?" and getting the first first answer
after boot. Normally, the board would have already been fully bootted
before getting OTAs.

You don't reallywhat communication methods are in play at each step (and
you should) but please restrict yourself to a single USB/serial connection
and not mixing network and serial.



> (I)
> (I) (ConnectToWiFi)(C1) Starting Web Server...
> (I) (begin)(C1) Connecting Web Endpoints
> (I) (begin)(C1) Embedded html file size: 1350
> (I) (begin)(C1) Embedded jsx file size: 41450
> (I) (begin)(C1) Embedded ico file size: 15406
> (I) (begin)(C1) Embedded timezones file size: 17834
> (I) (begin)(C1) HTTP server started
> (I) (ConnectToWiFi)(C1) Web Server begin called!
>
> You earlier said you'd cleared flash. Here, the flash of the user data is
present. It seems that someone somewhere has successfully completed a flash
of the user parition. Any theories when or how that happened?



>
> (I) (WriteCurrentEffectIndexFile)(C0) Number of bytes written to file /current.cfg: 1
> [ 11220][E][WiFiClient.cpp:517] flush(): fail on fd 54, errno: 11, "No more processes"
>
> This is troublesome. WriteCurrentEffectIndexFile should be writing just a
byte or two (one here). The write isn't failing, but the sync is. This
points to filesystem corruption in that second ("user", "secondary", or
"non-system") partition. We'd normally have basically reformatted that
partition if we'd tried to mount it and read it when we first booted.

So it's like the system has had a head injury and that second partition
both exists and doesn't exist. Fortunately, relatively little irretrievable
information relies on that (which effects were enabled/disabled and which
one is playing at this moment) so the heavy hand of erasing the partitions
and letting the system recreate that is not usually a terriblly distasteful
order fro teh doctors.

Personally, I'd attach it directly to serial and attempt to reinstall BOTH
the systems software ad "user data" partitions. Don't get OTA involved and
certainly not while this is running.

If you have to get microscopic, verify that you're getting the partiion
table dedicated to Mesmerizer. Dump flash at 0xc00 and confirm that the
bytes match those in config/partitions_custom_8M.csv

Beyond that making a friend in the group and comparing log files to see
where your device FIRST leaves the rails and comparing notest going through
a couple of COMPLETE flashing cycles is my best advice from here.

I don't think your board is physically broken.
Whether by cause (human or system) or by effect, I DO think the layout of
the "disk" that comprises the mesmerizer image (the system partition, the
partiion table, the OTA entries, the user parition, the respective file
systems and files, etc. is a bit scrambled. I think a bit of therapy can
probably get your system back to working.

Good luck!
RJL

**Reply by rbergen** [reply]
Just quickly commenting on a few of the log lines for completeness, because I've looked into them at some point in the past.

>> begin(): Failed starting MDNS
>> Error starting mDNS

> Now this, I don't remember seeing on eery boot. Can anyone else comment on whether they're seeing this one in a similar position?

Yes, this is normal. That is to say, I've not seen a single boot on my boards that doesn't include this.

>> (I) (ConnectToWiFi)(C1) Setting host name to NightDriverStrip...WL_NO_SHIELD

> Now THIS is concerning. This doesn't seem like an error, but is showing that src/network.cpp::ConnectToWiFi() is having trouble talking to the board.

I think the message indicates that the board tries to set its hostname before WiFi has actually connected. The hostname is not pulled in from DHCP.

>> (I) (ConnectToWiFi)(C1) Setting Clock...
>> [  7848][E][WiFiUdp.cpp:221] parsePacket(): could not receive data: 9

> This is the WiFi over the air updater getting data that's corrupt.

I believe this is the NTP client trying to parse data it doesn't yet have; `WiFiUDP::parsePacket()` is called in a loop in the NTP client code until it succeeds or times out.

>> (I) (WriteCurrentEffectIndexFile)(C0) Number of bytes written to file /current.cfg: 1
>> [ 11220][E][WiFiClient.cpp:517] flush(): fail on fd 54, errno: 11, "No more processes"

> This is troublesome. WriteCurrentEffectIndexFile should be writing just a byte or two (one here). The write isn't failing, but the sync is.

The flush actually concerns the WiFi client, not the filesystem. It's a bug in one of the dependencies that incorrectly thinks it _should_ have more data to push out over the network than it has available, and as such reports an error condition that isn't there. To add to the confusion, errno 11 actually stands for EAGAIN or "resource temporarily unavailable", which gets mistranslated to "No more processes" in this specific case. 
 This one took me a while to figure out when I looked into it.

**Comment by KeiranHines** [comment]
Thanks for all the info, to answer your questions:
> That's odd. The word "IoTWiFi" doesn't really appear in the code anywhere
that I can find. Do you have any theories where that name came from?
This board was provided to me pre-loaded from @davepl I suspect this is a network he had configured to test the device prior to shipping. My assumption was that that Wifi was stored in non-volatile memory and  persisted a standard flashing attempt. 

>  I know it's not the point, but are you also in Middle TN? It would be cool.
No I am in Australia, I use to live on a street that had a Simpsons reference, since then my IoT wifi has had a Simpsons reference. 

>Describe how you erased the flash and how you recovered from that. Did you
upload both the app and user partitions or just the app partition.
From the VS Code platform IO  plugin I selected the mesmerizer group, then platform -> Erase Flash. This executed `platformio.exe run --target erase --environment mesmerizer`

After erasing the flash and uploading again I got this. 
```
E (1195) esp_core_dump_flash: No core dump partition found!
E (1195) esp_core_dump_flas���r���ɕ�"յ�����ѥѥ���2�չ��jR�E (11) SPIFFS: mount failed, -10025
Replacing Idle Tasks with TaskManager...
(I) (PrintOutputHeader)(C1) NightDriverStrip
(I)
(I) (PrintOutputHeader)(C1) ------------------------------------------------------------------------------------------------------------
(I) (PrintOutputHeader)(C1) M5STICKC: 0, USE_M5DISPLAY: 0, USE_OLED: 0, USE_TFTSPI: 0, USE_LCD: 0, USE_AUDIO: 1, ENABLE_REMOTE: 1
(I) (PrintOutputHeader)(C1) ESP32 PSRAM Init: OK
(I) (PrintOutputHeader)(C1) Version 40: Wifi SSID: "Shelbyville" - ESP32 Free Memory: 235468, PSRAM:4191883, PSRAM Free: 4187123
(I) (PrintOutputHeader)(C1) ESP32 Clock Freq : 240 MHz
(I) (setup)(C1) Startup!
(I) (setup)(C1) Starting DebugLoopTaskEntry
>> Launching Debug Thread.  Mem: 235468, LargestBlk: 110580, PSRAM Free: 4187123/4191883, >> Launching JSON Writer Thread.  Mem: 226900, LargestBlk: 110580, PSRAM Free: 4187087/4191867, (W) (DeviceConfig)(C1) DeviceConfig could not be loaded from JSON, using defaults
(W) (NotifyJSONWriterThread)(C1) >> Notifying JSON Writer Thread
(W) (setup)(C1) Starting ImprovSerial
(W) (ReadWiFiConfig)(C1) Error (ESP_ERR_NVS_NOT_FOUND) opening NVS handle!
(W)
(W) (setup)(C1) Could not read WiFI Credentials
(W) (WriteWiFiConfig)(C1) Stored SSID and Password to NVS: Shelbyville, *******

Starting SmartMatrix Mallocs
Heap/32-bit Memory Available: 221600 bytes total, 110580 bytes largest free block
8-bit/DMA Memory Available  : 191304 bytes total, 110580 bytes largest free block
Total PSRAM used: 11328 bytes total, 4180459 PSRAM bytes free
SmartMatrix Layers Allocated from Heap:
Heap/32-bit Memory Available: 220124 bytes total, 110580 bytes largest free block
Starting SmartMatrix DMA Mallocs
sizeof framestruct: 00004000
DMA Memory Available before ptr1 alloc: 157028 bytes total, 110580 bytes largest free block
matrixUpdateFrames[0] pointer: 3FFD2B60
DMA Memory Available before ptr2 alloc: 157028 bytes total, 110580 bytes largest free block
matrixUpdateFrames[1] pointer: 3FFD6B70
Frame Structs Allocated from Heap:
Heap/32-bit Memory Available: 187324 bytes total, 110580 bytes largest free block
8-bit/DMA Memory Available  : 157028 bytes total, 110580 bytes largest free block
Total PSRAM used: 37312 bytes total, 4154347 PSRAM bytes free
Allocating refresh buffer:
lsbMsbTransitionBit of 0 requires 49152 RAM, 110580 available, leaving 61428 free: 
Raised lsbMsbTransitionBit to 0/7 to fit in RAM
lsbMsbTransitionBit of 0 gives 76 Hz refresh, 180 requested:
lsbMsbTransitionBit of 1 gives 152 Hz refresh, 180 requested: 
lsbMsbTransitionBit of 2 gives 300 Hz refresh, 180 requested:
Raised lsbMsbTransitionBit to 2/7 to meet minimum refresh rate
Descriptors for lsbMsbTransitionBit 2/7 with 16 rows require 12288 bytes of DMA RAM
SmartMatrix Mallocs Complete
Heap/32-bit Memory Available: 175004 bytes total, 110580 bytes largest free block
8-bit/DMA Memory Available  : 144708 bytes total, 110580 bytes largest free block
Total PSRAM used: 37312 bytes total, 4154347 PSRAM bytes free
Setting up parallel I2S bus at I2S1
Matrix Refresh Rate: 100
(I) (SetupBufferManagers)(C1) Could allocate 509 buffers but limiting it to 500
(I)
(W) (SetupBufferManagers)(C1) Reserving 500 LED buffers for a total of 3088000 bytes...
(I) (setup)(C1) Initializing splash effect manager...
(W) (InitSplashEffectManager)(C1) InitSplashEffectManager
(W) (InitEffectsManager)(C1) InitEffectsManager...
(I) (InitEffectsManager)(C1) Creating EffectManager using default effects
>> Launching Drawing Thread.  Mem: 173544, LargestBlk: 110580, PSRAM Free: 1010795/4174379, (W) (DrawLoopTaskEntry)(C1) >> DrawLoopTaskEntry
(W)
(W) (DrawLoopTaskEntry)(C1) Entering main draw loop!
>> Launching Audio Thread.  Mem: 169072, LargestBlk: 110580, PSRAM Free: 1010795/4174379, (I) (AudioSamplerTaskEntry)(C0) >>> Sampler Task Started
>> Launching Remote Thread.  Mem: 164376, LargestBlk: 110580, PSRAM Free: 1010627/4174363, (W) (begin)(C1) Remote Control Decoding Started
>> Launching Network Thread.  Mem: 158204, LargestBlk: 110580, PSRAM Free: 1010099/4174347, [ 20034][E][ESPmDNS.cpp:65] begin(): Failed starting MDNS
Error starting mDNS
(W) (UpdateSubscribers)(C1) Skipping Subscriber update, waiting for WiFi...
>> Launching ColorData Thread.  Mem: 149364, LargestBlk: 110580, PSRAM Free: 1010291/4174347, >> Launching Socket Thread.  Mem: 146428, LargestBlk: 110580, PSRAM Free: 1010291/4174347, (W) (NotifyJSONWriterThread)(C1) >> Notifying JSON Writer Thread
(I) (ConnectToWiFi)(C1) Setting host name to NightDriverStrip...WL_NO_SHIELD
(W) (ConnectToWiFi)(C1) Pass 1 of 5: Connecting to Wifi SSID: "Shelbyville" - ESP32 Free Memory: 108316, PSRAM:4174091, PSRAM Free: 997147
(W)
(W) (ColorDataTaskEntry)(C1) Started color data server!
(W) (SerializeWithBufferSize)(C0) Out of memory serializing object - increasing buffer to 4096 bytes
(W) (SerializeWithBufferSize)(C0) Out of memory serializing object - increasing buffer to 6144 bytes
[ 23402][E][vfs_api.cpp:182] remove(): /effects.cfg does not exists or is directory
(I) (SaveToJSONFile)(C0) Number of bytes written to JSON file /effects.cfg: 2883
(W) (NotifyJSONWriterThread)(C1) >> Notifying JSON Writer Thread
(I) (loop)(C1) WiFi: WL_CONNECTED, IP: 10.1.5.1, Mem: 105244, LargestBlk: 73716, PSRAM Free: 989179/4174043, LED FPS: 27 Refresh: 60 Hz, Power: 2252 mW, Brite: 100%, Audio FPS: 43, MinVU:  251.4, PeakVU:  259.0, VURatio: 0.1 Buffer: 0/500, CPU: 098%, 072%, FreeDraw: 0.011
(W) (ConnectToWiFi)(C1) Connected to AP with BSSID: <MY BSSID>
(W)
(W) (ConnectToWiFi)(C1) Received IP: 10.1.5.1
(I) (ConnectToWiFi)(C1) Starting/restarting Socket Server...
(W) (ProcessIncomingConnectionsLoop)(C1) Error accepting data!
(W) (SocketSToWiFi)(C1) Socket server started.
sed.  Retrying...
(I) 
I)
tToWiFi)(C1) Socket server started.
(I) (ConnectToWiFi)(C1) Publishing OTA...
(I) (ConnectToWiFi)(C1) Setting Clock...
[ 25489][E][WiFiUdp.cpp:221] parsePacket(): could not receive data: 9
(W) (UpdateClockFromWeb)(C1) NTP clock: Raw values sec=3812615827, usec=887693
(I) (UpdateClockFromWeb)(C1) Old Time: 25.800303, New Time: 1694238331.887693, Delta: 1694238306.087390
(I) (UpdateClockFromWeb)(C1) Adjusting time by 1694238306.087390 to 1694238331.887693
(I) (UpdateClockFromWeb)(C1) NTP clock: response received, updated time to: 1694238331.887693, DELTA: 1694238306.087390
(I)
(I) (ConnectToWiFi)(C1) Starting Web Server...
(I) (begin)(C1) Connecting Web Endpoints
(I) (begin)(C1) Embedded html file size: 1350
(I) (begin)(C1) Embedded jsx file size: 41450
(I) (begin)(C1) Embedded ico file size: 15406
(I) (begin)(C1) Embedded timezones file size: 17834
(I) (begin)(C1) HTTP server started
(I) (ConnectToWiFi)(C1) Web Server begin called!
[ 28234][E][vfs_api.cpp:182] remove(): /current.cfg does not exists or is directory
[ 28766][E][WiFiClient.cpp:517] flush(): fail on fd 54, errno: 11, "No more processes"
(I) (UpdateSubscribers)(C1) Got YouTube subscriber count for channel Daves Garage (GUID 9558daa1-eae8-482f-8066-17fa787bc0e4)
(I) (WriteCurrentEffectIndexFile)(C0) Number of bytes written to file /current.cfg: 1
(I) (loop)(C1) WiFi: WL_CONNECTED, IP: 10.1.5.1, Mem: 75492, LargestBlk: 40948, PSRAM Free: 985135/4173915, LED FPS: 35 Refresh: 60 Hz, Power: 1894 mW, Brite: 100%, Audio FPS: 43, MinVU:  350.6, PeakVU:  367.0, VURatio: 0.0 Buffer: 0/500, CPU: 096%, 084%, FreeDraw: 0.016
```
Similar results occurred on every reboot. 

For completeness, I went back to commit 12e941e, did a full clean, rebuild and upload. After that it works again. I went back to main, cleaned and rebuilt again and am now fully up to date and working. No idea why that fixed it, but all is well again. I can enjoy my Pong Clock having the correct time while I decide where I want to start developing new things. 

Thank-you for all your help.

**Comment by davepl** [comment]
To get back to “new”, you can do a command line

pio —erase_flash

(Or something like that, but it’s the pio erase flash command)

Then refresh the board.  You MIGHT have to hold down the BOOT button, but probably not.

Pretty sure the board itself is working, and I’ve never seen a matrix OR a board actually die.  Double-check your power connection and that you are perfectly centered up down with the PCB so as not to be off by one pin.

- Dave
